apps/prairielearn/src/pages/instructorFileTransfer/instructorFileTransfer.ts

```diff
@@ -37,6 +41,27 @@ async function getFileTransfer(file_transfer_id: string, user_id: string): Promi
   return file_transfer;
 }
 
+async function doTransfer(res: Response, editor: Editor, fileTransferId: string) {
+  const serverJob = await editor.prepareServerJob();
+  try {
+    await editor.executeWithServerJob(serverJob);
+  } catch {
+    res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+    return;
+  }
+
+  await sqldb.queryOneRowAsync(sql.soft_delete_file_transfer, {
+    id: fileTransferId,
+    user_id: res.locals.user.user_id,
+  });
+}
+
+function getContentFolder(fullPath: string, parentDir: string): string {
+  const path_exploded = path.normalize(fullPath).split(path.sep);
+  const content_dir_idx = path_exploded.findIndex((x) => x === parentDir);
+  return path_exploded.slice(content_dir_idx + 1).join(path.sep);
+}
+
```

Comment:
this is not new code, it is code lifted out of the handler so it can be used both for questions and course instances.